src/app/codelabs/all-exercises/all-exercises.component.ts

```diff
@@ -0,0 +1,51 @@
+import { ExerciseBase } from './../../exercise/exercise/exercise.base';
+import { Component } from '@angular/core';
+import { Ng2TsExercises, ng2tsConfig } from '../../../../ng2ts/ng2ts';
+
+@Component({
+  selector: 'slides-all-exercises',
+  templateUrl: './all-exercises.component.html',
+  styleUrls: ['./all-exercises.component.css']
+})
+export class AllExercisesComponent {
+  private exercise;
+  private milestones;
+  public allFiles = {};
+
+  constructor(private exercises: Ng2TsExercises) {
+    this.exercise = exercises.getExercises(1, 1);
+    this.milestones = ng2tsConfig.milestones.map((milestone) => {
+      return {
+        name: milestone.name,
+        exercises: milestone.exercises
+          .map((exercise, index) => { return { ...exercise, index }; })
+          .filter((exercise) => {
+            if (exercise['files']) { return true; }
+          })
+      };
+    });
+
+    this.milestones.forEach((milestone) => {
```

Comment:
Can you please add this to a private function for readability?

```
addAllExerciseFilesToFileMap(); // Easy to understand what you are trying to achieve

...

/**
 * Adds files to the allFiles object for easy look up (or whatever your reason)
 * This is optional since this might be pretty self explanatory 
 */
private addAllExerciseFilesToFileMap() {
...
}
```